pulsar-client-cpp/lib/ProducerImpl.cc

```diff
@@ -352,10 +353,16 @@ void ProducerImpl::sendAsync(const Message& msg, SendCallback callback) {
         }
         payload = encryptedPayload;
 
-        if (payloadSize > Commands::MaxMessageSize) {
-            LOG_DEBUG(getName() << " - compressed Message payload size" << payloadSize << "cannot exceed "
-                                << Commands::MaxMessageSize << " bytes");
-            cb(ResultMessageTooBig, msg);
+        if (cnx) {
+            if (payloadSize > cnx->getMaxMessageSize()) {
+                LOG_DEBUG(getName() << " - compressed Message payload size" << payloadSize << "cannot exceed "
+                                    << cnx->getMaxMessageSize() << " bytes");
+                cb(ResultMessageTooBig, msg);
+                return;
+            }
+        } else {
+            LOG_ERROR("Connection not ready for producer.")
```

Comment:
This is still returning error when we try to publish while not connected. Instead the producer must queue up the messages until it can reconnect (or the sending queue is full).